bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/device/CommandHandler.java

```diff
@@ -0,0 +1,912 @@
+     * 0x0B: set nontoggle on/off command
+     *
+     * crucially, the 0x09 command does not work (NACK from device)
+     *
+     * @author Bernd Pfrommer - openHAB 1 insteonplm binding
+     */
+    @NonNullByDefault
+    public static class LEDOnOffCommandHandler extends CommandHandler {
+        LEDOnOffCommandHandler(DeviceFeature f) {
+            super(f);
+        }
+
+        @Override
+        public void handleCommand(InsteonChannelConfiguration conf, Command cmd, InsteonDevice dev) {
+            try {
+                int button = this.getIntParameter("button", -1);
+                if (cmd == OnOffType.ON) {
+                    Msg m = dev.makeExtendedMessage((byte) 0x1f, (byte) 0x2e, (byte) 0x00,
+                            new byte[] { (byte) button, (byte) 0x09, (byte) 0x01 });
+                    dev.enqueueMessage(m, feature);
+                    logger.debug("{}: sent msg to switch {} on", nm(), dev.getAddress());
+                } else if (cmd == OnOffType.OFF) {
+                    Msg m = dev.makeExtendedMessage((byte) 0x1f, (byte) 0x2e, (byte) 0x00,
+                            new byte[] { (byte) button, (byte) 0x09, (byte) 0x00 });
+                    dev.enqueueMessage(m, feature);
+                    logger.debug("{}: sent msg to switch {} off", nm(), dev.getAddress());
+                }
+            } catch (IOException e) {
+                logger.warn("{}: command send i/o error: ", nm(), e);
+            } catch (FieldException e) {
+                logger.warn("{}: command send message creation error ", nm(), e);
+            }
+        }
+    }
+
+    @NonNullByDefault
+    public static class X10OnOffCommandHandler extends CommandHandler {
+        X10OnOffCommandHandler(DeviceFeature f) {
+            super(f);
+        }
+
+        @Override
+        public void handleCommand(InsteonChannelConfiguration conf, Command cmd, InsteonDevice dev) {
+            try {
+                byte houseCode = dev.getX10HouseCode();
+                byte houseUnitCode = (byte) (houseCode << 4 | dev.getX10UnitCode());
+                if (cmd == OnOffType.ON || cmd == OnOffType.OFF) {
+                    byte houseCommandCode = (byte) (houseCode << 4
+                            | (cmd == OnOffType.ON ? X10.Command.ON.code() : X10.Command.OFF.code()));
+                    Msg munit = dev.makeX10Message(houseUnitCode, (byte) 0x00); // send unit code
+                    dev.enqueueMessage(munit, feature);
+                    Msg mcmd = dev.makeX10Message(houseCommandCode, (byte) 0x80); // send command code
+                    dev.enqueueMessage(mcmd, feature);
+                    String onOff = cmd == OnOffType.ON ? "ON" : "OFF";
+                    logger.debug("{}: sent msg to switch {} {}", nm(), dev.getAddress(), onOff);
+                }
+            } catch (IOException e) {
+                logger.warn("{}: command send i/o error: ", nm(), e);
```

Comment:
This is actually here because Msg.makeMessage() throws an IOException if it can't find the message. This should never happen, except during development. Can I leave as is?

Reply:
done, created custom exception to replace IOException and left as warn.